enhancements/microshift/microshift-telemetry.md

```diff
@@ -0,0 +1,274 @@
+* Number of routes/ingress/services
+* Number of CRDs
+* OS version / Type (rpm / ostree)
+
+These metrics should be sent at least once a day.
+
+### Metrics details
+Based on [Telemetry API](https://rhobs-handbook.netlify.app/products/openshiftmonitoring/telemetry.md/)
+MicroShift will send metrics using Prometheus format. There are several ways
+of using the API: [using the client to forward metrics](https://github.com/openshift/telemeter/tree/main?tab=readme-ov-file#upload-endpoint-receive-metrics-in-client_modelmetricfamily-format-from-telemeter-client-currently-used-by-cmo),
+or [direct write requests](https://github.com/openshift/telemeter/tree/main?tab=readme-ov-file#metricsv1receive-endpoint-receive-metrics-in-prompbwriterequest-format-from-any-client).
+Using the client is not an option because that requires a local Prometheus
+instance, not viable in the typical resource constrained MicroShift deployment.
+The direct write requests take raw Prometheus data and require both special
+labels (`_id` for the cluster id) and crafting the authentication headers
+(cluster id and pull secret in a specific format in HTTP headers).
+
+OpenShift is using the same API and the same backend, so we need a way to
+distinguish MicroShift in the pool of metrics. For this we can use labels, as
+it does not require applying for new supported metrics. MicroShift metrics
+labels summary:
+| Label | Values |
+|:---|:---|
+|_id|cluster id|
+|label_kubernetes_io_arch|amd64, aarch64|
+|resource|Used to specify K8s resource names: pods, namespaces, etc.|
+|instance|ip address of the node|
+|version|microshift version|
+|ostree_commit|OStree commit id, if the system is deployed using ostree. This helps identifying fleets|
+
+Metrics from MicroShift are already supported in the API because OpenShift is
+using them. List follows:
+* cluster:capacity_cpu_cores:sum. Number of allocated CPUs for MicroShift.
+* cluster:capacity_memory_bytes:sum. Number of bytes of memory allocated for
+  MicroShift.
+* cluster:cpu_usage_cores:sum. Usage of CPU in percentage.
+* cluster:memory_usage_bytes:sum. Usage of memory in percentage.
+* cluster:usage:resources:sum. Usage of k8s resources, in count. Number of
+  pods, namespaces, services, etc.
+* cluster:usage:containers:sum. Number of active containers.
+* instance:etcd_object_counts:sum. Number of objects etcd contains.
+
+Combining metrics with their labels:
+| Metric | Labels |
+|:---|:---|
+|cluster:capacity_cpu_cores:sum|_id, instance, ostree_commit, version|
+|cluster:capacity_memory_bytes:sum|_id, instance, ostree_commit, version|
+|cluster:cpu_usage_cores:sum|_id, instance, ostree_commit|
+|cluster:memory_usage_bytes:sum|_id, instance, ostree_commit|
+|instance:etcd_object_counts:sum|_id, instance, ostree_commit|
+|cluster:usage:resources:sum|_id, instance, ostree_commit, resource|
+|cluster:usage:containers:sum|_id, instance, ostree_commit|
+
+Each metric, including a single sample and all labels, is under 250B in size.
```

Comment:
`cluster:usage:resources:sum` will have more than one sample (many `resource` label values).

Reply:
Included an estimation of sizing for each label and the whole write request. Also included a note about this metric having 6 different time series based on the resource.